odxtools/parameterinfo.py

```diff
@@ -20,12 +26,65 @@
 from .parameters.parameter import Parameter
 from .parameters.parameterwithdop import ParameterWithDOP
 from .parameters.reservedparameter import ReservedParameter
+from .parameters.systemparameter import SystemParameter
 from .parameters.tablekeyparameter import TableKeyParameter
 from .parameters.tablestructparameter import TableStructParameter
 from .paramlengthinfotype import ParamLengthInfoType
 from .staticfield import StaticField
 
 
+def _get_type_info(odx_type: DataType) -> str:
+    if odx_type == DataType.A_INT32:
+        return "int"
+    elif odx_type == DataType.A_UINT32:
+        return "uint"
+    elif odx_type in (DataType.A_FLOAT32, DataType.A_FLOAT64):
+        return "float"
+    elif odx_type == DataType.A_BYTEFIELD:
+        return "bytefield"
+    elif odx_type in (DataType.A_UNICODE2STRING, DataType.A_ASCIISTRING, DataType.A_UTF8STRING):
+        return "string"
+    else:
+        odxraise(f"Type info for type {odx_type.value}", NotImplementedError)
+        return "<unknown type>"
+
+
+def _get_linear_segment_info(segment: LinearSegment) -> str:
```

Comment:
I have a feeling this should be in the `__str__` or `__repr__` function of LinearSegment

Reply:
I don't think that `LinearSegment` and `RationalSegment` objects should be printed under normal circumstances, and if they are, usually something like the slope and the offset are desired? (IMO this is not the case in the context of `parameter_info()` were we are only interested in what ranges  can be specified  for a given parameter. Opinions welcome...)

Reply:
The idea is that the function that describes the object is human readable way, should be in the object itself, even if it's just for the sake of showing better values in the debugger.

Reply:
yes. my main point is that in most contexts other than `parameter_info()`, this probably requires a bit more information than just the domain where the segment is defined (i.e., some info what the segment does)...